### ISSUE TITLE
conda_list cannot search specified path properly, conda_binary cannot test path properly

### ISSUE THREAD

**Issue by xhdong-umd** [rstudio/reticulate#679]
I used rminiconda to install a miniconda, then hope to register the conda location with reticulate and use reticulate conda_ functions to install conda packages.

After setting `RETICULATE_MINICONDA_PATH` properly, conda_list can find it with auto option, but I want to specify it because there are multiple conda installations and the order is not guaranteed.

```r
> Sys.setenv(RETICULATE_MINICONDA_PATH = file.path(rminiconda::get_miniconda_path(), "gee"))
> reticulate:::miniconda_exists()
[1] TRUE
> conda_list()
  name                                          python
1  gee /Users/xhdong/Library/rminiconda/gee/bin/python
2  3.6      /Users/xhdong/anaconda/envs/3.6/bin/python
3 py27     /Users/xhdong/anaconda/envs/py27/bin/python
```

Then I tried to feed the conda location to conda_list to test it, then later use it to specify which conda to be used. However all kinds of variations cannot get it listed properly

```r
> conda_list("/Users/xhdong/Library/rminiconda/gee")
[1] name   python
<0 rows> (or 0-length row.names)
> conda_list("/Users/xhdong/Library/rminiconda")
[1] name   python
<0 rows> (or 0-length row.names)
> conda_list("/Users/xhdong/Library/rminiconda/gee/bin")
[1] name   python
<0 rows> (or 0-length row.names)
# check conda_binary, which doesn't give any meaningful result
> reticulate:::conda_binary("/Users/xhdong/Library/rminiconda/gee")
[1] "/Users/xhdong/Library/rminiconda/gee"
> reticulate:::conda_binary("/Users/xhdong/Library/rminiconda")
[1] "/Users/xhdong/Library/rminiconda"
```

I tracked and found [`conda_binary` logic have some missing cases](https://github.com/rstudio/reticulate/blob/master/R/conda.R#L234)

```r
altpath <- file.path(dirname(conda), "../bin/conda")
```
- line above try to go out of user provided path one step up, then go down one step down to bin to look for conda. 
- However dirname already provided a parent folder, so the altpath created will not find bin/conda, thus altpath is non-exist.
- Worse, the program leave this condition branch, but just test if the original path exist and return that path to be the conda binary path. 
   I'm not sure what's the case is expected here. If conda should always be either in bin/conda or condabin/conda.bat, is there any other valid case other than these two? line 244-250 means if a path didn't pass previous logic but exist, it will be reported as a valid conda binary.
   Here I think there is a confusion: is the function expecting the folder of conda exe (xx/bin), or the full path of conda executable (xx/bin/conda)? Normally user would expect the former, then just testing path exist doesn't really test if conda binary exist.

In summary
- `..` in line 234 should be removed
- the logic between 244-250 might need to be fully removed, other than an error report. 

@kevinushey , I'd love to create a PR if needed.

**Comment by xhdong-umd** [comment]
A PR created.
https://github.com/rstudio/reticulate/pull/680

**Comment by xhdong-umd** [comment]
Sorry, I realized the conda parameter is expecting the full path of conda binary, not the conda binary folder.
